### PR TITLE
Feature/152 arrow key navigation

### LINKED ISSUE #152: Arrow key navigation

### PR DESCRIPTION
**Pull request by Paulpanther**
Fixes #152

---
This change is [Reviewable](https://reviewable.io/reviews/hpi-swa-teaching/imapclient/186)

### CODE REVIEW

**Comment by Paulpanther** [comment]
More info about how it works in comments of #152

**Review by timgarrels — COMMENTED** [review]
I do approve of the changes and decisions made.
I have a change request on the `emailListKeyPress: aKey` function:
I do assume, that toolbuilder requires it's keyPress Method to return a bool, please confirm that.
The `emailListKeyPress: aKey` function should not contain a hardcoded handling by if Clause. That might be ok for the moment, as there is only one key to handle ($o), but in the future there might be a lot more.
I suggest to map keys to functions in a dictionary. The dictionary should be returned in a "static" instance method.
The current code in `emailListKeyPress` should then be moved to a block in that dictionary (or to a complete separate method, which might not be necessary here)

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPressMapping.st [hunk 1]
```diff
@@ -0,0 +1,6 @@
+events
+emailListKeyPressMapping
+
+	^ {
+		self class defaultOpenEmailChar -> [self selectedEmail ifNotNil: [self openEmail: self selectedEmail]]
```
Comment (felixrindt) [inline packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPressMapping.st]: maybe have the `$o` here directly so the functionality "Press o to open email" is visible in this line directly?!

**Review by felixrindt — CHANGES_REQUESTED** [review]

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPress..st [hunk 2]
```diff
@@ -0,0 +1,8 @@
+events
+emailListKeyPress: aKey
+	
+	(self emailListKeyPressMapping includesKey: aKey asLowercase) ifTrue: [
```
Comment (felixrindt) [inline packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPress..st]: Please use `at: key ifPresent: oneArgBlock ifAbsent: absentBlock` here :)

**Review by felixrindt — CHANGES_REQUESTED** [review]

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPress..st (hunk 2 above)
Comment (Paulpanther) [reply packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPress..st]: Done

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPressMapping.st (hunk 1 above)
Comment (Paulpanther) [reply packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListKeyPressMapping.st]: Done

**Review by felixrindt — APPROVED** [review]

**Pull request merged**